webhook/workspace/handler/immutable.go

```diff
@@ -177,7 +187,14 @@ func changePermitted(oldObj, newObj runtime.Object) bool {
 		log.Error(fmt.Errorf("Object %s is not a valid k8s object: does not have metadata", newObj.GetObjectKind()), "Failed to compare objects")
 		return false
 	}
-	newMeta.SetFinalizers(oldMeta.GetFinalizers())
-	newMeta.SetDeletionTimestamp(oldMeta.GetDeletionTimestamp())
-	return reflect.DeepEqual(oldMeta, newMeta)
+	oldLabels, newLabels := oldMeta.GetLabels(), newMeta.GetLabels()
+	oldAnnotations, newAnnotations := oldMeta.GetAnnotations(), newMeta.GetAnnotations()
```

Comment:
I would like to see variables declared nearest their usage.
BTW is there a need to have these vars?

Reply:
Mostly to avoid the clumsiness of the syntax `oldMeta.GetLabels()[constants.WorkspaceCreatorLabel]`